src/communication/utils.hpp

```diff
@@ -0,0 +1,74 @@
+#ifndef UTILS_HPP
```

Comment:
missing copyright header 
and I guess there could be other utils, so I would suggest communication_utils_hpp guard name